release/internal/hashreleaseserver/server.go

```diff
@@ -85,13 +85,10 @@ func remoteReleasesLibraryPath(user string) string {
 	return filepath.Join(RemoteDocsPath(user), "all-releases")
 }
 
-func HasHashrelease(hash string, cfg *Config) (bool, error) {
+func HasHashrelease(hash string, cfg *Config) bool {
 	logrus.WithField("hash", hash).Debug("Checking if hashrelease exists")
 	out, err := runSSHCommand(cfg, fmt.Sprintf("cat %s | grep %s", remoteReleasesLibraryPath(cfg.User), hash))
-	if err == nil {
-		return strings.Contains(out, hash), nil
-	}
-	return false, err
+	return err != nil && strings.Contains(out, hash)
```

Comment:
So now `HasHashrelease` returns `true` if we checked for a hash release and found one, and `false` if we didn't find one or if we had an error checking for one?

I see the logic there, but perhaps in addition we should log either that we had an error or that we found an existing hashrelease, just so we know which one happened to aid in debugging later? Something like:

```go
        if err != nil {
            logrus.WithField("hash", hash).Warn("Error checking hashrelease details") // Add error details here maybe
            return false
        }
        if strings.Contains(out, hash) {
            logrus.WithField("hash", hash).Info("Hashrelease already exists")
            return false
        }
	return true
```

Reply:
Should definitely be checking the error, and at a minimum logging it out.

Reply:
If we encounter an error, shouldn't we assume that the HR exists?

Seems like a better fix for this might be to make the command we run not exit with 1 on success... I think we're asking for future trouble if we're assuming "err != nil" is a success scenario.

Reply:
Right, I have updated the section to log errors and return the error when it is not expected